bin/mail/2016-03-29-ny-cs4all/generate-list.rb

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require 'active_support'
+require 'active_support/core_ext/object/blank'
+require 'set'
+
+# Gather all email addresses *@schools.nyc.gov who are (any of):
+# A. Code Studio teachers who have already been through the K-5 PD.
+# B. Code Studio teachers who have not been through any PD.
+# C. Hour of Code teachers who aren't on Code Studio.
+
+def query_dashboard_teachers(query)
+  {}.tap do |results|
+    DASHBOARD_DB.fetch(query).each do |teacher|
+      email = teacher[:email]
+      results[email] = {email: email, name: teacher[:name], id: teacher[:id]}
+    end
+  end
+end
+
+# A. Query Code Studio teachers who have already been through the K-5 PD.
+def query_k5_pd_teachers
+  section_ids = Set.new
```

Comment:
nit: Is there any advantage to making sections_ids a set rather than an array, noting that you cast to an array at the end? Also note that duplicates in the IN clause are fine.